Neos.Flow/Classes/ResourceManagement/ResourceManager.php

```diff
@@ -151,30 +153,21 @@ protected function initialize()
      */
     public function importResource($source, $collectionName = ResourceManager::DEFAULT_PERSISTENT_COLLECTION_NAME, $forcedPersistenceObjectIdentifier = null)
     {
-        $this->initialize();
-        if (!isset($this->collections[$collectionName])) {
-            throw new Exception(sprintf('Tried to import a file into the resource collection "%s" but no such collection exists. Please check your settings and the code which triggered the import.', $collectionName), 1375196643);
-        }
-
-        /* @var CollectionInterface $collection */
-        $collection = $this->collections[$collectionName];
-
-        try {
-            $resource = $collection->importResource($source);
-            if ($forcedPersistenceObjectIdentifier !== null) {
-                ObjectAccess::setProperty($resource, 'Persistence_Object_Identifier', $forcedPersistenceObjectIdentifier, true);
+        if (is_resource($source)) {
+            $content = stream_get_contents($source);
+            $filename = '';
+            if ($content === false) {
+                throw new Exception('Could not import the content stream as resource.', 1695390671);
             }
-            if (!is_resource($source)) {
-                $pathInfo = UnicodeFunctions::pathinfo($source);
-                $resource->setFilename($pathInfo['basename']);
+        } else {
+            $content = file_get_contents($source);
+            /** @var string $filename */
+            $filename = UnicodeFunctions::pathinfo($source, PATHINFO_FILENAME);
+            if ($content === false) {
+                throw new Exception(sprintf('Could not read the file from "%s" as resource.', $source), 1695390671);
```

Comment:
```suggestion
                throw new Exception(sprintf('Could not read the file from "%s" as resource.', $source), 1695390672);
```
Shouldn't this exception get it's own error code?